### ISSUE TITLE
Put in place infrastructure for translations

### ISSUE THREAD

**Issue by ars3niy** [ars3niy/tdlib-purple#27]

**Comment by BenWiederhake** [comment]
Mostly it's just about adding the appropiate commands to a special build target (maybe even outside of CMakeLists.txt), setting up an account on Transifex, uploading the potfile, inviting people, …

Actually, I think this is where I could make my first "real" contribution.

**Comment by ars3niy** [comment]
And implement it in the source code: right now I have `char *_(const char *s) { return (char *)s; }` :-) I've never worked with translations before you see, so don't even know what the library is called.

**Comment by BenWiederhake** [comment]
We used GNU intltool for telegram-purple. It's GPL'd, so my guess is that you're just fine with it as long as it works? :)

**Comment by ars3niy** [comment]
Definitely.

**Comment by BenWiederhake** [comment]
I'm still setting everything up, so don't jump to the translations yet! Here you can watch me setting everything up: https://www.transifex.com/tdlib-purple-developers/tdlib-purple/dashboard/

With a bit of luck, this will be the final URL.

Note that they first validate whether GPL really is OSI-approved (sigh), so this might take a while.

**Comment by ars3niy** [comment]
A few strings might yet change or disappear though

**Comment by BenWiederhake** [comment]
Sure, but setting everything up is the same amount of work either way, and right now I can take my time.

**Comment by ars3niy** [comment]
The translations (or rather, at the moment, the translation) works, and is included in binary packages from now on.

**Comment by BenWiederhake** [comment]
Whee!

So in theory one could bring in translators now: I would just write an e-mail on the telegram-translators public mailing list (like I did for telegram-purple), and an announcement on the telegram-purple Transifex project page (to hijack the old translators).

However, I'd first like to go over the source and provide a bit more context for each string, like "pop-up window title" or "error message" or "buddy status" or "must not translate the word 'kick'!" or something along the lines. This way, the translators will know the context for each phrase when translating. I know that I had a hard time when translating it to German, and probably mistranslated some things as a result.

tl;dr: I wanna do certain things before we pull in the translator volunteer army.

**Comment by BenWiederhake** [comment]
Roadmap:
- [X] Add hints for the translators (#74)
- [ ] Check that all large text is in a `_(…)`
- [ ] Support pluralization (it's pretty simple, but I have to read it again)
- [ ] In strings with multiple objects ("{} posted in {}, size: {}") that might change order in some languages, show how to re-order them (I guess "foo {0} bar {1} baz" should work?)
- [ ] Maybe unify _Accept/_Cancel and _Accept/_Cancel?
- [ ] Unify Yes/No and _Yes/_No, as well as OK/Cancel and _OK/_Cancel
